src/api/app/lib/backend/proxy_request.rb

```diff
@@ -0,0 +1,136 @@
+module Backend
+  # The backend shares (a subset of) our API routes. This is a service to proxy requests to those routes.
+  # It caches incoming/outgoing files in our tmp directory or makes use of the various forward/rewrite/redirect
+  # plugins for apache, nginx or lighttpd if configured.
+  class ProxyRequest
+    attr_accessor :path, :request
+
+    def initialize(request, path_override = nil)
+      @request = request
+      @path = path_override || path_from_request
+    end
+
+    def call
+      case request.method_symbol
+      when :get, :head
+        proxy_from_backend unless use_web_server_forward
+      when :post, :put, :delete
+        proxy_to_backend
+      end
+    end
+
+    private
+
+    def path_from_request
+      path = @request.path_info
+      query_string = @request.query_string
+      if @request.form_data?
+        # it's uncommon, but possible that we have both
+        query_string += '&' if query_string.present?
+        query_string += @request.raw_post
+      end
+      query_string = '?' + query_string if query_string.present?
+      path + query_string
+    end
+
+    def use_web_server_forward
```

Comment:
do we have to support different servers here?
And if yes, cannot we extract it to a lib/gem?

Reply:
> do we have to support different servers here?

Yes.

> And if yes, cannot we extract it to a lib/gem?

Not really, it depends on our configuration.

Reply:
so could you at least extract to some lib? This code is irrelevant here.

Reply:
How is it irrelevant? It's part of proxying. Either we proxy via file or we proxy via the plugins.